typescript/infra/scripts/secret-rpc-urls/set-rpc-urls.ts

```diff
@@ -0,0 +1,105 @@
+import { confirm } from '@inquirer/prompts';
+import { ethers } from 'ethers';
+
+import {
+  getSecretRpcEndpoints,
+  getSecretRpcEndpointsLatestVersionName,
+  secretRcpEndpointsExist,
+  setSecretRpcEndpoints,
+} from '../../src/agents/index.js';
+import { disableGCPSecretVersion } from '../../src/utils/gcloud.js';
+import { getArgs, withChain, withRpcUrls } from '../agent-utils.js';
+
+async function testProviders(rpcUrlsArray: string[]): Promise<boolean> {
+  let providersSucceeded = true;
+  for (const url of rpcUrlsArray) {
+    const provider = new ethers.providers.StaticJsonRpcProvider(url);
+    try {
+      await provider.getBlockNumber();
```

Comment:
@tkporter - I think you may have mentioned comparing whether the block numbers match. Have rpcs been wildly different in the past?

If we do this, there is a very small chance that they could be different organically with a new block.

Reply:
Prompting the user to confirm after the tests!